ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/LedgerApiTestTool.scala

```diff
@@ -39,16 +41,38 @@ object LedgerApiTestTool {
   private def exitCode(summaries: Vector[LedgerTestSummary], expectFailure: Boolean): Int =
     if (summaries.exists(_.result.isLeft) == expectFailure) 0 else 1
 
-  private def printAvailableTests(): Unit = {
-    println("Tests marked with * are run by default.")
+  private def printAvailableTests(config: Config): Unit = {
+    val session = new LedgerSession(
+      LedgerSessionConfiguration(
+        config.participants,
+        config.shuffleParticipants,
+        config.tlsConfig,
+        config.loadScaleFactor,
+        config.partyAllocation,
+      ),
+      new ParticipantSessionManager
+    )(ExecutionContext.global)
+    def print_suites(suites: Map[String, LedgerSession => LedgerTestSuite]): Unit = {
```

Comment:
Thanks for updating `--list`!

Reply:
It does feel quite messy but I don't see how to do better without a massive restructuring. The current structure has just not been designed to extract test names without running them.